src/braket/default_simulator/cirq/program_context.py

```diff
@@ -0,0 +1,84 @@
+from typing import List, Tuple
+
+import cirq
+import numpy as np
+from braket.ir.jaqcd.program_v1 import Results
+from cirq import Circuit
+
+from braket.default_simulator.openqasm.program_context import AbstractProgramContext
+
+# cirq.XX, cirq.YY, and cirq.ZZ gates are not the same as Braket gates
+CIRQ_GATES = {
+    "i": cirq.I,
+    "h": cirq.H,
+    "x": cirq.X,
+    "y": cirq.Y,
+    "z": cirq.Z,
+    "cnot": cirq.CNOT,
+    "cz": cirq.CZ,
+    "s": cirq.S,
+    "t": cirq.T,
+    "cphaseshift": cirq.cphase,
+    "rx": cirq.rx,
+    "ry": cirq.ry,
+    "rz": cirq.rz,
+    "swap": cirq.SWAP,
+    "iswap": cirq.ISWAP,
+    "ccnot": cirq.CCNOT,
+    "cswap": cirq.CSWAP,
+    "ccz": cirq.CCZ,
+    "ccx": cirq.CCX,
+    "measure": cirq.MeasurementGate,
+}
+
+
+class ProgramContext(AbstractProgramContext):
+    def __init__(self):
+        super().__init__(Circuit())
+
+    def is_builtin_gate(self, name: str) -> bool:
+        user_defined_gate = self.is_user_defined_gate(name)
+        return name in CIRQ_GATES and not user_defined_gate
+
+    def add_phase_instruction(self, target: Tuple[int], phase_value: int):
+        raise NotImplementedError
+
+    def _gate_accepts_parameters(self, gate_class):
+        return hasattr(gate_class, "parameters") and len(gate_class.parameters) > 0
+
+    def add_gate_instruction(
+        self, gate_name: str, target: Tuple[int], params, ctrl_modifiers: List[int], power: int
+    ):
+        qubits = [cirq.LineQubit(int(qubit)) for qubit in target]
+        target_qubits = qubits[len(ctrl_modifiers) :]
+        control_qubits = qubits[: len(ctrl_modifiers)]
+
+        if params:
+            gate = CIRQ_GATES[gate_name](*params).on(*target_qubits)
+        else:
+            gate = CIRQ_GATES[gate_name].on(*target_qubits)
```

Comment:
would this line work the same as `gate = CIRQ_GATES[gate_name]().on(*target_qubits)`?

If so, maybe we can remove the if/else and add a line `params = params or []` and then just do `gate = CIRQ_GATES[gate_name](*params).on(*target_qubits)`

Reply:
I was thinking to do it exactly like you said.

The problem is gates with params have __init__ which accepts angle. And other gates which use different gates to define the gate are already initialized. 

Objects have __call__ method which accepts qubits.

Hence I decided to go with if condition. Let me know if you think we can find a way around this to just write a single line.

Reply:
this should be fine then